src/main/java/org/zalando/nakadi/service/StorageService.java

```diff
@@ -87,24 +105,33 @@ public StorageService(final ObjectMapper objectMapper,
 
         try {
             storageDbRepository.createStorage(storage);
-        } catch (DuplicatedStorageIdException | InternalNakadiException e) {
-            return Result.problem(e.asProblem());
+        } catch (final RepositoryProblemException e) {
+            LOG.error("DB error occurred when creating storage", e);
+            return Result.problem(Problem.valueOf(INTERNAL_SERVER_ERROR, e.getMessage()));
+        }
+        catch (final DuplicatedStorageException e) {
+            return Result.problem(Problem.valueOf(CONFLICT, e.getMessage()));
         }
-
         return Result.ok();
     }
 
     public Result<Void> deleteStorage(final String id) {
         try {
-            if (!storageDbRepository.getStorage(id).isPresent()) {
-                return Result.notFound("No storage with ID " + id);
-            }
-            if (storageDbRepository.isStorageUsed(id)) {
-                return Result.forbidden("Storage " + id + " is in use");
-            }
-            storageDbRepository.deleteStorage(id);
-        } catch (InternalNakadiException e) {
+            transactionTemplate.execute(action -> {
+                storageDbRepository.deleteStorage(id);
+                return null;
+            });
+        } catch (final NoStorageException e) {
```

Comment:
I would love to do that but this PR already grew too much. But later we of course should remove this Result.